harvard-university-for-the-creative-arts.csl

```diff
@@ -167,20 +203,25 @@
           </group>
         </group>
       </else-if>
-      <else-if type="map" match="any">
-        <group delimiter=" ">
-          <text variable="genre" prefix="[" suffix="]"/>
-          <text variable="scale" suffix="."/>
-          <text variable="publisher-place" suffix=":"/>
-          <text variable="publisher" suffix="."/>
-          <text variable="collection-title" prefix="(" suffix=")"/>
-        </group>
-      </else-if>
       <else>
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher" suffix="."/>
-        </group>
+        <choose>
+          <if match="any" variable="publisher" type="motion_picture broadcast report musical_score song entry-dictionary entry-encyclopedia">
+            <text variable="publisher" suffix="."/>
+          </if>
+          <else>
+            <text value="s.n." prefix="(" suffix=")."/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="any" variable="publisher-place" type="broadcast motion_picture report musical_score song entry-dictionary entry-encyclopedia manuscript">
+        <text variable="publisher-place"/>
+      </if>
```

Comment:
@ibadger, you can test this yourself, but I'm not sure this will work correctly. With `match="any"`, if-statement will report true whenever either a given item has a value for "publisher-place", or if it's of one of the item types you test for with `type`. Is that the logic you're after? E.g. a report without a value for "publisher-place" won't print "s.l.".